native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/AbstractNativeMojo.java

```diff
@@ -123,78 +123,112 @@ protected AbstractNativeMojo() {
     }
 
     protected boolean isMetadataRepositoryEnabled() {
-        return metadataRepositoryConfiguration != null && metadataRepositoryConfiguration.isEnabled();
+        return metadataRepositoryConfiguration == null || metadataRepositoryConfiguration.isEnabled();
     }
 
     protected void configureMetadataRepository() {
-        if (isMetadataRepositoryEnabled()) {
-            Path repoPath = null;
-            Path destinationRoot = reachabilityMetadataOutputDirectory.toPath();
-            if (Files.exists(destinationRoot) && !Files.isDirectory(destinationRoot)) {
-                throw new RuntimeException("Metadata repository must be a directory, please remove regular file at: " + destinationRoot);
-            }
+        if (!isMetadataRepositoryEnabled()) {
+            logger.warn("GraalVM reachability metadata repository is disabled");
+            return;
+        }
+
+        Path destinationRoot = reachabilityMetadataOutputDirectory.toPath();
+        if (Files.exists(destinationRoot) && !Files.isDirectory(destinationRoot)) {
+            throw new RuntimeException("Metadata repository must be a directory, please remove regular file at: " + destinationRoot);
+        }
+        try {
+            Files.createDirectories(destinationRoot);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        Path repoPath = metadataRepositoryConfiguration != null ? getRepo(destinationRoot) : getDefaultRepo(destinationRoot);
+        if (repoPath == null) {
+            throw new RuntimeException("Cannot pull GraalVM reachability metadata repository either from the one specified in the configuration or the default one.\n" +
+                    "Note: Since the repository is enabled by default, you can disable it manually in your pom.xml file (see this: " +
+                    "https://graalvm.github.io/native-build-tools/latest/maven-plugin.html#_configuring_the_metadata_repository)");
+        } else {
+            metadataRepository = new FileSystemRepository(repoPath, new FileSystemRepository.Logger() {
+                @Override
+                public void log(String groupId, String artifactId, String version, Supplier<String> message) {
+                    logger.info(String.format("[graalvm reachability metadata repository for %s:%s:%s]: %s", groupId, artifactId, version, message.get()));
+                }
+            });
+        }
+    }
+
+    private Path getDefaultRepo(Path destinationRoot) {
+        // try to get default Metadata Repository from Maven Central
+        URL targetUrl = resolveDefaultMetadataRepositoryUrl();
+        if (targetUrl == null) {
+            logger.warn("Unable to find the GraalVM reachability metadata repository in Maven repository. " +
+                    "Falling back to the default repository.");
```

Comment:
Would be nice to show the fallback URL as part of the warning message.

Reply:
We can do that, but we are also printing URL in the repository download step ([here](https://github.com/graalvm/native-build-tools/pull/557/files/02eb0699a7d38f8c2ab58a5b18770aa7836db958#diff-45f2dd0466adde99b21b98a4313076b0ad20c41980f2107382eecbde2d9996a2R226)). If we are printing it later, is it necessary to print it here as well?